forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-mock-store/forgerock-openbanking-uk-aspsp-rs-mock-store-server/src/main/java/com/forgerock/openbanking/aspsp/rs/store/repository/migration/v3_1_6/MongoDbPaymentsChangeLog.java

```diff
@@ -24,83 +24,216 @@
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRDomesticScheduledConsentMigrator.toFRDomesticScheduledConsent;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRDomesticScheduledPaymentSubmissionMigrator.toFRDomesticScheduledPaymentSubmission;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRDomesticStandingOrderConsentMigrator.toFRDomesticStandingOrderConsent;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRDomesticStandingOrderPaymentSubmissionConverter.toFRDomesticStandingOrderPaymentSubmission;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRFileConsentMigrator.toFRFileConsent;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRFilePaymentSubmissionMigrator.toFRFilePaymentSubmission;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRInternationalConsentMigrator.toFRInternationalConsent;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRInternationalPaymentSubmissionMigrator.toFRInternationalPaymentSubmission;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRInternationalScheduledConsentMigrator.toFRInternationalScheduledConsent;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRInternationalScheduledPaymentSubmissionMigrator.toFRInternationalScheduledPaymentSubmission;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRInternationalStandingOrderConsentMigrator.toFRInternationalStandingOrderConsent;
-import static com.forgerock.openbanking.aspsp.rs.store.repository.migration.v3_1_6.payments.FRInternationalStandingOrderPaymentSubmissionMigrator.toFRInternationalStandingOrderPaymentSubmission;
-
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.data.util.Pair;
+import org.springframework.util.StopWatch;
+
+import java.beans.IntrospectionException;
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/*
+ * Data base migration changeSet
+ * Describe what changes need to be done for objects with version v3_1_6
+ * For that case we will change the dataType from the field 'countrySubDivision' from Array to String
+ * Fix of issue: https://github.com/OpenBankingToolkit/openbanking-toolkit/issues/27, ticket: 59553
+ */
 @ChangeLog
 @Slf4j
 public class MongoDbPaymentsChangeLog {
 
```

Comment:
The ChangeLog class expected by mongobee, this class execute the datatype upgrade process using bulk operations.
Only will be upgraded the documents version 3.1.6.

Reply:
The obVersion filter has been deleted, it means that we will update all documents.